src/CauserResolver.php

```diff
@@ -0,0 +1,131 @@
+<?php
+namespace Spatie\Activitylog;
+
+use Closure;
+use Illuminate\Auth\AuthManager;
+use Illuminate\Contracts\Config\Repository;
+use Illuminate\Database\Eloquent\Model;
+use InvalidArgumentException;
+use Spatie\Activitylog\Exceptions\CouldNotLogActivity;
+
+class CauserResolver
+{
+    protected AuthManager $authManager;
+
+    protected string $authDriver;
+
+    /**
+     * User defined callback to override default reslover logic
+     */
+    protected Closure | null $resolverOverride = null;
+
+    /**
+     * User defined model to override default reslover logic
+     */
+    protected Model | null $causerOverride = null;
+
+    public function __construct(Repository $config, AuthManager $authManager)
+    {
+        $this->authManager = $authManager;
+
+        $this->authDriver = $config['activitylog']['default_auth_driver'] ?? $this->authManager->getDefaultDriver();
+    }
+
+    /**
+     * Reslove causer based different arguments first we'll check for override closure
+     * Then check for the result causer if it valid. In other case will return the
+     * override causer defined by the user or delgate to the getCauser() method
+     *
+     * @param Model|int|null $subject
+     * @return null|Model
+     * @throws InvalidArgumentException
+     * @throws CouldNotLogActivity
+     */
+    public function resolve(Model | int | null $subject = null) : ?Model
```

Comment:
Also here we should support UUIDs

```suggestion
    public function resolve(Model | int | string | null $subject = null) : ?Model
```